module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,59 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+#include <sstream>
+
+CompressedImgPixelInserter&
+CompressedImgPixelInserter::operator=(const Pixel pixel) {
+    if (columnIdx_ == 0 or (pixel != img_.back().first)) {
+        img_.emplace_back(std::make_pair(pixel, 1));
```

Comment:
Did you know that you can make pair without calling std::make_pair? 
Like this: 
img_.emplace_back(pixel, 1);

Reply:
@WBigaj is it possible to create pair like this? I know that `img_.emplace_back(pixel, 1);` is possible, but I never heard about `img_.emplace_back(pixel, 1);`

Reply:
@Kasperice  This is exactly what `empalce_back` do. It calls directly constructor with args. So when you use `empace_back` and use aslo `std::make_pair` you make it wrong. Because you create temporary variable. 
So when You want use `make_pair` use `push_back`. And when You want directly call constructor use `emplace_back` with arguments.

Reply:
Please change to `push_back` or use `empalce_back` in proper way.

Reply:
@nauka-programowania-MA thank you for clarification :)

Reply:
It's exactly like @nauka-programowania-MA says. BTW there is thread about this in stackoverflow:
https://stackoverflow.com/questions/53900865/emplace-back-vs-push-back-when-inserting-a-pair-into-stdvector